src/Basescape/PurchaseState.cpp

```diff
@@ -159,12 +175,39 @@ PurchaseState::PurchaseState(Base *base, CannotReequipState *parent) : _base(bas
 		_cats.push_back("STR_FILTER_MISSING");
 	}
 
-	RuleBaseFacilityFunctions providedBaseFunc = _base->getProvidedBaseFunc({});
+	auto providedBaseFunc = _base->getProvidedBaseFunc({});
```

Comment:
Please leave the explicit type name (RuleBaseFacilityFunctions).
I try to use auto only for pointers, references, lambdas, etc.
For simple ints, bools, etc. I try to avoid auto.
I also try to avoid auto when a copy is made; or where there is a chance of any confusion; or lack of specificity.
See 9a90fab4372841b387319900993d0267251db18b